go/mysql/binlog/binlog_json_test.go

```diff
@@ -14,69 +14,92 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package json
+package binlog
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql/json"
 )
 
+func jsonObject(entries map[string]*json.Value) *json.Value {
+	var obj json.Object
+	for k, v := range entries {
+		obj.Set(k, v, json.Set)
+	}
+	return json.NewObject(obj)
+}
+
 func TestBinaryJSON(t *testing.T) {
```

Comment:
We are testing nested documents already in the parser:
```
		{
			name:     `nested object {"asdf":{"foo":123}}`,
			data:     []byte{0, 1, 0, 29, 0, 11, 0, 4, 0, 0, 15, 0, 97, 115, 100, 102, 1, 0, 14, 0, 11, 0, 3, 0, 5, 123, 0, 102, 111, 111},
			expected: `JSON_OBJECT(_utf8mb4'asdf', JSON_OBJECT(_utf8mb4'foo', 123))`,
		},
```

I also added a couple more values to the list in the e2e tests. I think we have enough coverage for these.
```
		`[1, "abc", -1, 0.2342342342349, {"a":"b","c":"d","ab":"abc","bc":["x","y"]}]`,
		`{"a":2947293482093480923840923840923, "cba":{"a":2947293482093480923840923840923, "cba":334234234234234234234234234.234234239090}}`,
```

Reply:
Agreed that the parser is really where we want to extensively cover this since those are also very cheap to run tests that are much easier to iterate on than an end-to-end test.